File: app/src/test/java/c_Classes_and_Objects/ArtistTest.java
package c_Classes_and_Objects;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;

import c_Classes_and_Objects.f_Data_Classes.Artist;

public class ArtistTest {

	private static final int ID = 1;
	private static final int NEW_ID = 2;

	private Artist artist;

	@Before
	public void setup() {
		artist = new Artist(ID, "name", "url", "mbid");
	}

	@Test
	public void when_instantiated_thenSettersStillWork() {
		//GIVEN
		assertThat(artist.getId()).isEqualTo(ID);
		//WHEN
		artist.setId(NEW_ID);
		//THEN
		assertThat(artist.getId()).isEqualTo(NEW_ID);
	}
}

File: app/src/test/java/c_Classes_and_Objects/ArtistTest.kt
package c_Classes_and_Objects

import c_Classes_and_Objects.f_Data_Classes.Artist
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class ArtistTest {

    private var artist = Artist(ID.toLong(), "name", "url", "mbid")

    @Test
    fun when_instantiated_thenSettersStillWork() {
        assertThat(artist.id).isEqualTo(ID.toLong())

        artist.id = NEW_ID

        assertThat(artist.id).isEqualTo(NEW_ID)
    }

    companion object {

        private const val ID = 1
        private const val NEW_ID : Long = 2
    }
}
